### PR TITLE
Allow all headers to be set as Response headers

### PR DESCRIPTION
**Pull request by StefH**
Related to:

https://github.com/WireMock-Net/WireMock.Net/issues/137
https://github.com/WireMock-Net/WireMock.Net/issues/136
https://github.com/WireMock-Net/WireMock.Net/issues/126

### CODE REVIEW

**Comment by StefH** [comment]
@alastairtree can you review this pr?

**Comment by alastairtree** [comment]
I'm not sure I understand the notifiable for this change? Any background?

**Comment by StefH** [comment]
Related to:

https://github.com/WireMock-Net/WireMock.Net/issues/137
https://github.com/WireMock-Net/WireMock.Net/issues/136
https://github.com/WireMock-Net/WireMock.Net/issues/126

File: test/WireMock.Net.Tests/FluentMockServerTests.Proxy.cs [hunk 1]
```diff
@@ -190,29 +193,30 @@ public partial class FluentMockServerTests
         [Fact]
         public async Task FluentMockServer_Proxy_Should_change_absolute_location_header_in_proxied_response()
         {
-            // given
-            _serverForProxyForwarding = FluentMockServer.Start();
+            // Assign
+            var settings = new FluentMockServerSettings { AllowPartialMapping = false };
+            _serverForProxyForwarding = FluentMockServer.Start(settings);
```
Comment (alastairtree) [inline test/WireMock.Net.Tests/FluentMockServerTests.Proxy.cs]: Perhaps names these 2 servers more specifically as it is quite hard to follow. _serverThatReturnsRedirects and serverThatProxies ?

File: test/WireMock.Net.Tests/FluentMockServerTests.Proxy.cs [hunk 2]
```diff
@@ -190,29 +193,30 @@ public partial class FluentMockServerTests
...
             _serverForProxyForwarding
                 .Given(Request.Create().WithPath("/*"))
                 .RespondWith(Response.Create()
                     .WithStatusCode(HttpStatusCode.Redirect)
                     .WithHeader("Location", _serverForProxyForwarding.Urls[0] + "testpath"));
```
Comment (alastairtree) [inline test/WireMock.Net.Tests/FluentMockServerTests.Proxy.cs]: This creates a server that could redirect indefinitely I think. `Request.Create().WithPath("/")` instead?

File: test/WireMock.Net.Tests/FluentMockServerTests.cs [hunk 3]
```diff
@@ -364,24 +367,25 @@ public void FluentMockServer_Admin_Mappings_Add_SameGuid()
                 new object[] { new JsonPathMatcher("$..[?(@.message == 'Hello server')]"), "text/plain" }
             };
 
-        [Theory]
```
Comment (alastairtree) [inline test/WireMock.Net.Tests/FluentMockServerTests.cs]: Any particular reason for not using the parameterised tests?

File: test/WireMock.Net.Tests/FluentMockServerTests.Proxy.cs (hunk 1 above)
Comment (StefH) [reply test/WireMock.Net.Tests/FluentMockServerTests.Proxy.cs]: I understand your comment, however for me it's clear for now. But I will think about this to see if I want to rename the these variables.

File: test/WireMock.Net.Tests/FluentMockServerTests.Proxy.cs (hunk 2 above)
Comment (StefH) [reply test/WireMock.Net.Tests/FluentMockServerTests.Proxy.cs]: Fixed

Comment (StefH) [reply test/WireMock.Net.Tests/FluentMockServerTests.Proxy.cs]: No, it has to be like this, else test fails.

**Comment by StefH** [comment]
Thanks a lot!
